### PR TITLE
Add pre-commit hook

### PR DESCRIPTION
**Pull request by AlessioC31**
Added `pre-commit` git hook and instructions on how to activate it.
I decided to not use `just` in the pre-commit hook since it may be that developers not use it.

### CODE REVIEW

File: extra/precommit-script.sh [hunk 1]
```diff
@@ -0,0 +1,15 @@
+#/bin/sh
```
Comment (guerinoni) [inline extra/precommit-script.sh:1]: `#! /bin/sh`

File: extra/precommit-script.sh [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+cargo clippy --workspace -- -D warnings         \
+    -W clippy::complexity                       \
+    -W clippy::correctness                      \
+    -W clippy::nursery                          \
+    -W clippy::perf                             \
+    -W clippy::style                            \
+    -W clippy::suspicious
+
+cargo test --workspace
```
Comment (guerinoni) [inline extra/precommit-script.sh:15]: Missing end empty line :)

File: extra/precommit-script.sh (hunk 1 above)
Comment (AlessioC31) [reply extra/precommit-script.sh:1]: > `#! /bin/sh`

Whoops, thanks

**Review by guerinoni — APPROVED** [review]
LGTM

**Pull request merged**